### PR TITLE
Support _FILE based configuration values

### LINKED ISSUE #253: Support _FILE based configuration values for all config keys
To configure this tool, a lot of potentially sensitive and powerful credentials and tokens need to be passed. Right now, these have to be passed as env vars. Docker Swarm supports "secrets" which allows putting credentials into files made available to the container at runtime.

This was added for S3 credentials in https://github.com/offen/docker-volume-backup/pull/161, but not for anything else.

Unfortunately package `envconfig` which is used here does not (and will not) support this out of the box: https://github.com/kelseyhightower/envconfig/issues/130

Supporting this pattern for all values would be a very nice addition though, so I can see a few ways forward:
- find out if there is a fork of package `envconfig` that supports this (it's mostly dormant anyways)
- fork package `envconfig` and add the functionality ourselves
- find a different package for handling configuration (this would need to be non-breaking)

### PR DESCRIPTION
**Pull request by MaxJa4**
Closes #253

## Changes

All environment variables can now be used with the `_FILE` convention.
This means, that e.g. `WEBDAV_PASSWORD` can also be supplied via docker secrets by using `WEBDAV_PASSWORD_FILE` instead.

**Note**: All _FILE variants are _preferred_ over their non-file variant.

## Todo

- [x] README update
- [x] Tests

### CODE REVIEW

**Comment by MaxJa4** [comment]
@m90 used mostly your code from #253 for the lookup function and pointed the `envconfig` module to our fork for now. Will open a PR to the original repo soon. I'd wait a little, maybe it gets merged rather soon, so we can change `offen/envconfig` back to how it was before in the `go.mod`/imports.

I also experimented with the `--timeout` option for docker compose, as DVB always gets forcibly killed by compose after the default 10s timeout. Reason [seems](https://stackoverflow.com/questions/63447441/docker-stop-for-crond-times-out) to be, that `crond` does not properly implement the SIGINT signal, so it doesn't react to the container stop. Since containers are recreated and stopped a lot in the tests, this should speed them up a bit without causing any issues.
I used 3 seconds here, since containers usually stop almost immediately and they get recreated anyway. That reduced the CI time from ~16:30 min. to ~13:00 min. Since we're adding more and more tests, this can maybe help us a little to not someday need 30 mins just für CI tests. This change shouldn't be added here though, it's just a test that I'll remove again later. Maybe you want to include that in your #261 PR about the docker/testing setup :)

**Comment by m90** [comment]
This looks like a great addition, thanks.

> I'd wait a little, maybe it gets merged rather soon, so we can change offen/envconfig back to how it was before in the go.mod/imports.

Ok, just let me know when you're done waiting. Or not.

> I also experimented with the --timeout option for docker compose,

Good idea, I'll add these to #261

**Comment by MaxJa4** [comment]
Created the PR to the source repo: https://github.com/kelseyhightower/envconfig/pull/210
Merged our changes into `offen/envconfig@master` and created a new `v1.5.0` version for Go to target.

File: test/swarm/run.sh [hunk 1]
```diff
@@ -29,6 +29,7 @@ sleep 5
 expect_running_containers "5"
 
 docker stack rm test_stack
+sleep 1
```
Comment (m90) [inline test/swarm/run.sh:32]: Does this test fail reliably without this sleep or did you just happen to see it fail intermittently? I think we can keep it, I'd just like to understand it.

File: cmd/backup/script.go [hunk 2]
```diff
@@ -89,6 +89,18 @@ func newScript() (*script, error) {
 		return nil
 	})
 
+	envconfig.Lookup = func(key string) (string, bool) {
+		location, ok := os.LookupEnv(key + "_FILE")
```
[the remark this answers is not in the document]
Comment (MaxJa4) [reply cmd/backup/script.go:93]: I see the probability of people having a random, stray `_FILE` variant floating around (specifically for the backup container) as very very low.

I checked out the `postgres` Docker image source ([entrypoint](https://github.com/docker-library/postgres/blob/master/16/alpine3.18/docker-entrypoint.sh)) and found this:

```sh
file_env() {
	local var="$1"
	local fileVar="${var}_FILE"
	local def="${2:-}"
	if [ "${!var:-}" ] && [ "${!fileVar:-}" ]; then
		printf >&2 'error: both %s and %s are set (but are exclusive)\n' "$var" "$fileVar"
		exit 1
	fi
	local val="$def"
	if [ "${!var:-}" ]; then
		val="${!var}"
	elif [ "${!fileVar:-}" ]; then
		val="$(< "${!fileVar}")"
	fi
	export "$var"="$val"
	unset "$fileVar"
}
```

TLDR:
- If both variants are present -> panic
- Prefer the value over the _FILE variant

Makes sense to me, partially because that's the `old` way of doing it and you can see the contents of the variable right in the configuration file, not indirectly in e.g. Docker secrets.

File: cmd/backup/script.go [hunk 3]
```diff
@@ -89,6 +89,18 @@ func newScript() (*script, error) {
...
+		if ok {
+			contents, err := os.ReadFile(location)
+			if err != nil {
```
[the remark this answers is not in the document]
Comment (MaxJa4) [reply cmd/backup/script.go:96]: Since (I think) we just agreed in the last comment on preferring the value over the _FILE variant, it would mean that the user specifically wants to use a _FILE and that failed -> undefined/undesired behavior for the user -> panic on any error.

File: test/swarm/run.sh (hunk 1 above)
Comment (MaxJa4) [reply test/swarm/run.sh:32]: This was an intermittent fault, probably a timing issue. Didn't run into this before, just added the second of sleeping to hopefully not run into such a timing issue there again.

File: cmd/backup/script.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (m90) [reply cmd/backup/script.go:93]: Sounds good to me as well, yes.

**Comment by MaxJa4** [comment]
Done from my side :)

**Review by m90 — APPROVED** [review]
:100:
